src/aura/BGE_TemplateDetails/BGE_TemplateDetailsController.js

```diff
@@ -0,0 +1,98 @@
+        component.set('v.templateMetadata', templateMetadataView);
+        var availableTemplateFieldsView = helper.AvailableTemplateFieldsView(component, model);
+        component.set('v.availableTemplateFields', availableTemplateFieldsView);
+        var activeTemplateFieldsView = helper.ActiveTemplateFieldsView(component, model);
+        component.set('v.activeTemplateFields', activeTemplateFieldsView);
+
+        model.init();
+    },
+
+    /* ***************************************************************
+     * @Description. Cancels edit or creation of the template.
+     *****************************************************************/
+    cancel: function(component, event, helper) {
+
+    },
+
+    /* ***************************************************************
+     * @Description. Saves the template information. 
+     *****************************************************************/
+    save: function(component, event, helper) {
+        var templateInfoData = component.get("v.templateInfo");
+        var model = component.get('v.model');
+        model.getTemplateInfo().load(templateInfoData);
+        model.save();
+    },
+
+    /******************************** Template Fields Controller Functions *****************************/
+
+    /* ***************************************************************
+     * @Description. Updates the selected available fields in the 
+     * Template Fields Model.
+     *****************************************************************/
+    selectAvailableTemplateFields: function (component, event, helper) {
+        var selectedFields = {};
+        event.getParam('selectedRows').forEach(function(selectedRow) {
+            selectedFields[selectedRow.id] = selectedRow;
+        });
+        var model = component.get('v.model');
+        model.getTemplateFields().selectAvailables(selectedFields);
+    },
+
+    /* ***************************************************************
+     * @Description. Updates the selected active fields in the 
+     * Template Fields Model.
+     *****************************************************************/
+    selectActiveTemplateFields: function (component, event, helper) {
+        var selectedFields = {};
+        event.getParam('selectedRows').forEach(function(selectedRow) {
+            selectedFields[selectedRow.id] = selectedRow;
+        });
+        var model = component.get('v.model');
+        model.getTemplateFields().selectActives(selectedFields);
+    },
+
+    /* ***************************************************************
+     * @Description. Updates the selected available fields in the 
+     * Template Fields Model to active.
+     *****************************************************************/
+    updateToActive: function (component, event, helper) {
+        var model = component.get('v.model');
+        model.getTemplateFields().updateToActive();
+    },
+
```

Comment:
Same here. Can this and the following be combined?

Reply:
In this case it might be possible to pass in some parameter to give context for where the action is coming from, but that might double function length anyways. I'd probably prefer to keep them separate for clarity.